### ISSUE TITLE
Outdated explanations due to changes starting in React Router 4.0.0

### ISSUE THREAD

**Issue by Gua-naiko-che** [tayiorbeii/egghead.io_idiomatic_redux_course_notes#25]
Hi,

I'm following the course right now and I found a problem in lesson 5 "Adding React Router to the project". After making the changes to introduce Router and Route, I was getting this error:

`_this.props.history is undefined`

After searching a bit, I found [this answer](https://stackoverflow.com/a/44263018/2651069) explaining that there has been a change starting in React Router 4.0.0, and that we should use BrowserRouter from react-router-dom, instead of Router from react-router. I have tried this solution and the app works as expected. Here is the fixed code:
```

import React, { PropTypes } from 'react';
import { Provider } from 'react-redux';
import App from './App';
import { BrowserRouter, Route } from 'react-router-dom'

const Root = ({ store }) => (
    <Provider store={store}>
        <BrowserRouter>
            <Route path="/" component={App} />
        </BrowserRouter>
    </Provider>
);
```

Do you think we should add this information to the notes? Or at list explain that we should use a previous version of react-router for it to work as in the video?

**Comment by Gua-naiko-che** [comment]
Other problems arising from the changes in React Router 4.0.0:

In [06. Navigating with React Router](https://github.com/tayiorbeii/egghead.io_idiomatic_redux_course_notes/blob/master/06-Navigating_with_React_Router_Link.md#06-navigating-with-react-router-link), the optional argument should have this syntax:
`<Route path="/:filter?" component={App} />`
instead of:
`<Route path="/(:filter)" component={App} />`
[Source](https://stackoverflow.com/a/35604855/2651069)



And to [rewrite the FilterLink component](https://github.com/tayiorbeii/egghead.io_idiomatic_redux_course_notes/blob/master/06-Navigating_with_React_Router_Link.md#filterlinkjs-after), we should use `NavLink `instead of `Link`, and the `to `property needs to add a '/' to the filter. This:
```
const FilterLink = ({ filter, children }) => (
    <NavLink
        exact
        to={'/' + (filter === 'all' ? '' : filter)}
        activeStyle={{
            textDecoration: 'none',
            color: 'black',
        }}
    >
        {children}
    </NavLink>
);
```
instead of this:
```
const FilterLink = ({ filter, children }) => (
  <Link
    to={filter === 'all' ? '' : filter}
    activeStyle={{
      textDecoration: 'none',
      color: 'black',
    }}
  >
    {children}
  </Link>
);
```
[Source](https://reacttraining.com/react-router/web/api/NavLink)

**Comment by tayiorbeii** [comment]
@Gua-naiko-che Thanks for recording this. Please feel free to open a PR appending these notes in the proper locations :)

**Comment by Gua-naiko-che** [comment]
Done 😄 
https://github.com/tayiorbeii/egghead.io_idiomatic_redux_course_notes/pull/26

**Comment by tayiorbeii** [comment]
Closed via #26
